actors/evm/src/lib.rs

```diff
@@ -173,6 +176,48 @@ impl EvmContractActor {
         let output = RawBytes::from(exec_status.output_data.to_vec());
         Ok(output)
     }
+
+    pub fn bytecode<BS, RT>(rt: &mut RT) -> Result<Cid, ActorError>
+    where
+        BS: Blockstore + Clone,
+        RT: Runtime<BS>,
+    {
+        // Any caller can fetch the bytecode of a contract; this is now EXT* opcodes work.
+        rt.validate_immediate_caller_accept_any()?;
+
+        let state: State = rt.state()?;
+        Ok(state.bytecode)
+    }
+
+    pub fn storage_at<BS, RT>(rt: &mut RT, params: GetStorageAtParams) -> Result<U256, ActorError>
+    where
+        BS: Blockstore + Clone,
+        RT: Runtime<BS>,
+    {
+        // This method cannot be called on-chain; other on-chain logic should not be able to
+        // access arbitrary storage keys from a contract.
+        rt.validate_immediate_caller_is([&fvm_shared::address::Address::new_id(0)])?;
+
+        let state: State = rt.state()?;
+        let blockstore = rt.store().clone();
+
+        // load the storage HAMT
+        let mut hamt =
+            Hamt::<_, _, U256>::load(&state.contract_state, blockstore).map_err(|e| {
+                ActorError::illegal_state(format!(
+                    "failed to load storage HAMT on invoke: {e:?}, e"
+                ))
+            })?;
+
+        let mut system = System::new(rt, &mut hamt).map_err(|e| {
+            ActorError::unspecified(format!("failed to create execution abstraction layer: {e:?}"))
+        })?;
```

Comment:
Looks like we need to do this a lot. Helper?

Reply:
we should probably make one, but that's out of scope here.

I would like to do a general cleanup/beatify pass at some point.

Reply:
btw, feel free to go wild and apply bigclaw magic in the code when there is a lull.